url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/logic/ItemLogic.java

```diff
@@ -0,0 +1,622 @@
+			Scroll_Suf newsuf) {
+		List<Item> newitems = new ArrayList<>();
+
+		for (Shoes x : items.getShoes()) {
+			Shoes changed = new Shoes(x);
+			boolean success = changed.brynn(newpre, newsuf);
+			if (success) {
+				newitems.add(changed);
+				shoes_changed.add(x);
+			}
+		}
+		for (Pants x : items.getPants()) {
+			Pants changed = new Pants(x);
+			boolean success = changed.brynn(newpre, newsuf);
+			if (success) {
+				newitems.add(changed);
+				pants_changed.add(x);
+			}
+		}
+		for (Gloves x : items.getGloves()) {
+			Gloves changed = new Gloves(x);
+			boolean success = changed.brynn(newpre, newsuf);
+			if (success) {
+				newitems.add(changed);
+				gloves_changed.add(x);
+			}
+		}
+		for (Armor x : items.getArmors()) {
+			Armor changed = new Armor(x);
+			boolean success = changed.brynn(newpre, newsuf);
+			if (success) {
+				newitems.add(changed);
+				armors_changed.add(x);
+			}
+		}
+		for (Helm x : items.getHelms()) {
+			Helm changed = new Helm(x);
+			boolean success = changed.brynn(newpre, newsuf);
+			if (success) {
+				newitems.add(changed);
+				helms_changed.add(x);
+			}
+		}
+
+		return newitems;
+	}
+
+	public void filterSetNames(Items items, String ignore) {
+		if (ignore.length() == 0) {
+			return;
+		}
+		String setname;
+
+		List<String> names = new ArrayList<String>();
+
+		if (!ignore.contains(",")) {
+			names.add(ignore);
+		} else {
+			String[] igar = ignore.split(",");
+			for (String x : igar) {
+				names.add(x.trim());
+			}
+		}
+
+		int count = 0;
+		List<Shoes> sho = new ArrayList<>();
+		List<Pants> pan = new ArrayList<>();
+		List<Gloves> glo = new ArrayList<>();
+		List<Armor> arm = new ArrayList<>();
+		List<Helm> hel = new ArrayList<>();
+		for (String igname : names) {
+			for (Shoes s : items.getShoes()) {
+				setname = s.getItemset().getName();
+				if (setname.toLowerCase().contains(igname)) {
+					sho.add(s);
+					count++;
+				}
+			}
+			for (Pants s : items.getPants()) {
+				setname = s.getItemset().getName();
+				if (setname.toLowerCase().contains(igname)) {
+					pan.add(s);
+					count++;
+				}
+			}
+			for (Gloves s : items.getGloves()) {
+				setname = s.getItemset().getName();
+				if (setname.toLowerCase().contains(igname)) {
+					glo.add(s);
+					count++;
+				}
+			}
+			for (Armor s : items.getArmors()) {
+				setname = s.getItemset().getName();
+				if (setname.toLowerCase().contains(igname)) {
+					arm.add(s);
+					count++;
+				}
+			}
+			for (Helm s : items.getHelms()) {
+				setname = s.getItemset().getName();
+				if (setname.toLowerCase().contains(igname)) {
+					hel.add(s);
+					count++;
+				}
+			}
+		}
+		items.getShoes().removeAll(sho);
+		items.getPants().removeAll(pan);
+		items.getHelms().removeAll(hel);
+		items.getArmors().removeAll(arm);
+		items.getGloves().removeAll(glo);
+		System.out.println(count + " Items removed due to ignore ( " + ignore
+				+ " )");
+
+	}
+
+	public void filterPrice(Items items, double price) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 279-298 into a separate method.